src/index.ts

```diff
@@ -217,25 +303,37 @@ export class GraphQLClient {
   }
 
   /**
-   * Send a GraphQL document to the server.
+   * Send GraphQL documents in batch to the server.
    */
   async batchRequests<T extends any = any, V = Variables>(
     documents: BatchRequestDocument<V>[],
     requestHeaders?: Dom.RequestInit['headers']
+  ): Promise<T>
+  async batchRequests<T = any, V = Variables>(options: BatchRequestOptions): Promise<T>
+  async batchRequests<T = any, V = Variables>(
+    arg1: BatchRequestDocument<V>[] | BatchRequestOptions,
+    arg2?: Dom.RequestInit['headers']
```

Comment:
Better parameter names?

Reply:
The current generic approach is fine with me (inspired by React Query [useQuery](https://github.com/tannerlinsley/react-query/blob/5fd537e4c2ad39aa5f36d96736d8426228d16c42/src/react/useQuery.ts#L42)) but we can rename like below if you prefer:
- `arg1` => `documentsOrOptions`
- `arg2` => `requestHeaders`

Reply:
@jasonkuhrt Ok to rename or we keep as is?

Reply:
Let's rename

Reply:
Done in https://github.com/prisma-labs/graphql-request/pull/303/commits/c513e619769644e121ea7b9a0feebb2185c38401